File: Tutoriels/Python/Projets/src/manager/GConfig.py
#================================================
class GConfig:
    #================================================
    m_instance = None
    #================================================
    def __init__(self):
        if GConfig.m_instance == None:
            self.m_dataMap = {}
            GConfig.m_instance = self        
    #================================================
    @staticmethod
    def Instance():
        if GConfig.m_instance == None:
            GConfig()
        return GConfig.m_instance
    #================================================
    def setData(self, key, data):
        self.m_dataMap[key] = data      
    #================================================
    def getData(self, key):
        return self.m_dataMap[key]     
    #================================================
    def showData(self):
        for key, value in self.m_dataMap.items():
            print(key + " = " + value)
        print
#================================================

File: Tutoriels/CSharp/Projets/src/manager/GConfig.cs
//===============================================
using System;
using System.Collections.Generic;
//===============================================
public class GConfig {
    //===============================================
    private static GConfig m_instance = null;
    private Dictionary<string, string> m_dataMap;
    //===============================================
    private GConfig() {
        m_dataMap = new Dictionary<string, string>();
    }
    //===============================================
    public static GConfig Instance() {
        if(m_instance == null) {
            m_instance = new GConfig();
        }
        return m_instance;
    }
    //===============================================
    public void SetData(string key, string data) {
        m_dataMap[key] = data;
    }
    //===============================================
    public string GetData(string key) {
        return m_dataMap[key];
    }
    //===============================================
    public void ShowData() {
        foreach (KeyValuePair<string, string> lData in m_dataMap) {
            Console.Write("{0} = {1}\n", lData.Key, lData.Value);
        }
        Console.Write("\n");
    }
    //===============================================
}
//===============================================
